cpp/rrtmgp_conversion.h

```diff
@@ -5,36 +5,81 @@
 #include <netcdf.h>
 #include <stdexcept>
 
+// Validate if both enabled?
 #ifdef RRTMGP_ENABLE_KOKKOS
+#ifdef RRTMGP_ENABLE_YAKL
+// Both are on, validate
+#define COMPUTE_SWITCH(yimpl, kimpl) \
+  kimpl; RRT_REQUIRE(yimpl == kimpl, "Bad COMPUTE_SWITCH")
```

Comment:
This kind of assumes that the macro is uses as
```
blah = COMPUTE_SWITCH (kimpl, yimpl);
```
You could maybe be more generic, and do
```
RRT_REQUIRE ( (kimpl)==(yimpl) ), (kimpl)
```
The comma operator evaluates the first operand, discards the result, and then evaluates and returns the second.

Note: I wrapped macro args within parentheses, since that avoid any potential issue with operator precedence, but maybe that's not needed.

Reply:
The impl you provided did not work, but I did add parens where I could.